ethereum/run_test_guardian.py

```diff
@@ -0,0 +1,486 @@
+    # Give them 1600 ether
+    genesis.set_storage(ETHER, addr, amount)
+
+
+# Initialize the first guardians
+for i, key_hex in enumerate(genesis_config['guardians']):
+    k = key_hex.decode('hex')
+    # Generate the address
+    a = ecdsa_accounts.privtoaddr(k)
+    assert big_endian_to_int(genesis.get_storage(a, 2**256 - 1)) == 0
+    # Give them 1600 ether
+    #genesis.set_storage(ETHER, a, 1600 * 10**18)
+    # Give them A LOT more ether
+    genesis.set_storage(ETHER, a, 1000 * 1600 * 10**18)
+    # Make their validation code
+    vcode = ecdsa_accounts.mk_validation_code(k)
+    print 'Length of validation code:', len(vcode)
+    # Make the transaction to join as a Casper guardian
+    txdata = casper_ct.encode('join', [vcode])
+    tx = ecdsa_accounts.mk_transaction(0, 25 * 10**9, 1000000, CASPER, 1500 * 10**18, txdata, k, True)
+    print 'Joining'
+    v = tx_state_transition(genesis, tx, listeners=[my_listen])
+    index = casper_ct.decode('join', ''.join(map(chr, v)))[0]
+    print 'Joined with index', index
+    print 'Length of account code:', len(get_code(genesis, a))
+    # Check that the EVM that each account must have at the end
+    # to get transactions included by default is there
+    assert mandatory_account_evm == get_code(genesis, a).rstrip('\x00')
+    # Check sequence number
+    assert big_endian_to_int(genesis.get_storage(a, 2**256 - 1)) == 1
+    # Check that we actually joined Casper with the right
+    # validation code
+    vcode2 = call_method(genesis, CASPER, casper_ct, 'getGuardianValidationCode', [index])
+    assert vcode2 == vcode
+
+
+# Determine the number of nodes that will be run.
+#
+# TODO: This will need to be reworked so that we can initialize the genesis
+# state without needing to know about which other nodes are participating.
+#
+KEY_IDX = get_arg('--key-idx', int, None)
+KEY = get_arg('--key', str, None)
+
+if KEY is None:
+    if KEY_IDX is None:
+        KEY = zpad(encode_int(random.randint(1e16, 1e32)), 32)
+        print "Using KEY=", KEY.encode('hex')
+    elif KEY_IDX < len(genesis_config['alloc']):
+        KEY = genesis_config['alloc'][KEY_IDX][0].decode('hex')
+    else:
+        raise Exception("Key index out of range")
+elif KEY_IDX is not None:
+    raise Exception("Cannot specify both --key and --key-idx")
+else:
+    KEY = zpad(KEY, 32)
+
+
+JOIN_AT_BLOCK = get_arg('--join-at-block', int, -1)
+
+
+GENESIS_TIMESTAMP = genesis_config['timestamp']
+
+
+# Set the starting RNG seed to equal to the number of casper guardians
+# in genesis
+genesis.set_storage(RNGSEEDS, encode_int32(2**256 - 1), genesis.get_storage(CASPER, 0))
+# Set the genesis timestamp
+genesis.set_storage(GENESIS_TIME, encode_int32(0), GENESIS_TIMESTAMP)
+print 'genesis time', GENESIS_TIMESTAMP, '\n' * 10
+# Create betting strategy objects for every guardian
+# TODO: may not need to `clone()` this anymore since it's singular.
+bet = defaultBetStrategy(genesis.clone(), KEY, join_at_block=JOIN_AT_BLOCK)
```

Comment:
Yes this `clone()` should be avoided. It's actually cloned a new `OverlayDB` because no new_dbfile is given.